docs/design/proposals/resource-reference/resource-reference.md

```diff
@@ -0,0 +1,555 @@
+## Introduction
+
+This proposal discusses adding the support for referencing an ACK custom resource
+in another custom resource. The reconciler will wait for the referenced custom
+resource to be created, and in a stable state, before proceeding to create the
+resource which references it.
+
+For example, currently if ACK customer needs to create an APIGatewayv2 `Integration`,
+they first need to
+1. create an `API` custom resource
+2. wait for `API` resource creation to finish
+3. manually find the `APIID` from `Status` of `API` resource
+4. create an `Integration` resource with the `APIID`
+
+Having to manually wait for the resource creation prevents declarative configuration
+techniques that rely on describing all desired resources at the same time using
+one or more Kubernetes YAML manifests. But with approach of referenced resources,
+ACK controller will wait for `API` resource to get created and then use the API
+reference to find `APIID` and create Integration resource.
+
+The deletion strategy for the referenced resource is out of scope for this proposal
+right now. For Ex: If API resource is referenced in integration resource, the deletion
+of API resource will not be blocked. We plan to handle deletion strategy in future.
+
+## Scope
+
+### First Revision
+* Allow adding resource name(s) from same api-group as reference in the custom
+resource definition
+* Only the resources from same namespace will be allowed to use as reference
+* Only the fields of type '*string' or '[]*string' can be referenced from
+another resource. This satisfies the use case of referencing 'ID(s)/Name(s)/ARN(s)'
+from other resources
+
+### Future Improvements
+* Add support for custom hooks while dereferencing resources
+* Add support for `selector` for referencing the objects
+* Add support for references in custom fields that are not present in
+* Add support for referencing resource from different api-groups
+* Add support for not deleting a resource if it is referenced in another resource
+* Add support for finite retries and terminal conditions when resolving references
+
+## Preferred Approach
+
+### High Level Overview
+
+* Introduce new construct in generator.yaml which will indicate to code-generator
+if a new spec field with type `AWSResourceReferenceWrapper` or
+`AWSResourcesReferenceWrapper` should be added to the custom resource definition.
```

Comment:
I don't think we'd need a `AWSResourcesReferenceWrapper` struct. This would just be `[]*AWSResourceReferenceWrapper`, no?